ciphers/caesar_cipher.cpp

```diff
@@ -0,0 +1,123 @@
+/**
+ * @file caesar_cipher.cpp
+ * @brief Implementation of [Caesar cipher](https://en.wikipedia.org/wiki/Caesar_cipher) algorithm.
+ *
+ * @details
+ *      In cryptography, a Caesar cipher, also known as Caesar's cipher, the shift cipher, 
```

Comment:
The doxygen generator uses Markdown syntax formatting. Hence, starting a paragraph with spaces renders the text as `pre` and is not right at all. Please see the output of one of your previous contribution: 
![image](https://user-images.githubusercontent.com/7001608/87861652-368e9980-c916-11ea-9d02-90bfe1ccbf2b.png)
```suggestion
 * In cryptography, a Caesar cipher, also known as Caesar's cipher, the shift cipher, 
 * Caesar's code or Caesar shift, is one of the simplest and most widely known encryption 
 * techniques. It is a type of substitution cipher in which each letter in the plaintext 
 * is replaced by a letter some fixed number of positions down the alphabet. For example, 
 * with a left shift of 3, D would be replaced by A, E would become B, and so on. 
 * The method is named after Julius Caesar, who used it in his private correspondence.
 *
 * ### Algorithm
 * The encryption can also be represented using modular arithmetic by first transforming 
 * the letters into numbers, according to the scheme, A → 0, B → 1, ..., Z → 25.
 * Encryption of a letter x by a shift n can be described mathematically as,
 * \f[ E(x) = (x + n)\;\mbox{mod}\; 26\f]
 * while decryption can be described as,
 * \f[ D(x) = (x - n) \;\mbox{mod}\; 26\f]
```

Reply:
> Please see the output of one of your previous contribution:

I can change comments and make a new pull request for correcting ```bogo_sort```. Should I do it ?

Reply:
sure. That'd be great, thank you 😄